src/Aspire.Hosting/ApplicationModel/ResourceCommandAnnotation.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
+/// <summary>
+/// Represents a command annotation for a resource.
+/// </summary>
+[DebuggerDisplay("Type = {GetType().Name,nq}, Source = {Source}, Target = {Target}")]
+public sealed class ResourceCommandAnnotation : IResourceAnnotation
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ResourceCommandAnnotation"/> class.
+    /// </summary>
+    public ResourceCommandAnnotation(string type, string displayName, Func<CustomResourceSnapshot, Task<bool>> visible, Func<string, IServiceProvider, Task> invokeCommand, string? iconContent, bool isHighlighted)
+    {
+        ArgumentNullException.ThrowIfNull(type);
+        ArgumentNullException.ThrowIfNull(displayName);
+
+        Type = type;
+        DisplayName = displayName;
+        Visible = visible;
+        InvokeCommand = invokeCommand;
+        IconContent = iconContent;
+        IsHighlighted = isHighlighted;
+    }
+
+    /// <summary>
+    /// Gets the source of the bind mount or name if a volume. Can be <c>null</c> if the mount is an anonymous volume.
+    /// </summary>
+    public string Type { get; }
```

Comment:
Consider a different name for this property, perhaps simply `Name`. As I understand, this is used as a key/ID in the set of commands available for a particular resource. `Type` suggests there can be more than one command with the same value of this property.

Reply:
I agree that name would be better than type. Type is already used in the proto API between the host and dashboard.

Reply:
OK